### DISCUSSION TITLE
Poll: what to do with streams?

### DISCUSSION THREAD

**Discussion by vweevers** [Level/community#105] [General]
In a world without `levelup` (https://github.com/Level/community/issues/58), what would you like to do with [streams](https://github.com/Level/levelup#dbcreatereadstreamoptions)?

_Edit: poll is closed (2021-12-10)._

| Option |  Votes
|:--------|:------
| Remove in favor of `for await...of db.iterator()` | 5
| Move to module: `streams(db); db.createReadStream()` | 3
| Move to module: `createReadStream(db)` | 2
| Use `level-iterator-stream`: `stream(db.iterator())` | 1
| Add to `abstract-leveldown` (temporarily to ease migration) |  1
| Add to `abstract-leveldown` (using `streamx`) | 1
| Add to `abstract-leveldown` |  0

**Comment by MeirionHughes** [comment]
Personally, I've been favouring WebAPI Streams. So I've already been wrapping iterators with my own code to make them API ReadableStream.  Given the examples above, mine would be comparable to `createReadStream(db, opts)`. 

If I'm pragmatic - adding it to abstract-leveldown is the path of least resistance and least breaking changes, while achieving your goal of getting rid of levelup.

**Comment by juliangruber** [comment]
I'd like to propose an option that isn't on this list: use async iteration, on iterators

**Reply by vweevers** [reply]
@juliangruber What do you mean by async iteration, if not `for await...of`? That's the last option on the list, and supported [today](https://github.com/Level/abstract-leveldown#for-awaitof-iterator).

**Reply by juliangruber** [reply]
Ah, I thought the last option was for async iteration on `db.create*` methods, I meant `for await ... of db.iterator()` yeah!

**Reply by vweevers** [reply]
Edited to clarify that.

**Comment by vweevers** [comment]
I closed the poll. Thanks for voting! Most folks are in favor of removing streams. Some folks still want to use them, so I will write a standalone module (tracked in https://github.com/Level/abstract-level/issues/2 until it has its own repo) and an upgrade guide for folks that will be migrating from a `levelup` db.

**Comment by vweevers** [comment]
To follow up: you can now choose between [`for await...of`](https://github.com/Level/abstract-level#for-awaitof-iterator) and [Node.js streams](https://github.com/Level/read-stream) and [Web Streams](https://github.com/Level/web-stream).
